src/app/services/baw-api/baw-api-state.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from "@angular/core";
+import { AuthToken } from "@interfaces/apiInterfaces";
+import { User } from "@models/User";
+import { BehaviorSubject, Observable } from "rxjs";
+
+export const guestUser = undefined;
+export const guestAuthToken = undefined;
+type GuestUser = typeof guestUser;
+type GuestAuthToken = typeof guestAuthToken;
+
+export interface AuthTriggerData {
+  user: User | GuestUser;
+  authToken?: AuthToken | GuestAuthToken;
+}
+
+@Injectable()
+export class BawApiStateService {
```

Comment:
```suggestion
export class BawSessionState {
```

We don't need _state_ and _service_ in a name.